### ISSUE TITLE
Better generalizability for the reorder-columns operation

### ISSUE THREAD

**Issue by wetneb** [OpenRefine/OpenRefine#5576]
We offer an operation, `reorder-columns`, which is able to change the order of columns in an arbitrary way and remove any number of columns, in a single step.

![image](https://user-images.githubusercontent.com/309908/213440782-41467e32-c841-424a-bb17-6a049a61691e.png)

The operation which corresponds to this screenshot is represented internally as:
```json
  {
    "op": "core/column-reorder",
    "columnNames": [
      "Identifiant du lieu",
      "Année du tournage",
      "Type de tournage",
      "Titre",
      "director",
      "producer",
      "Réalisateur",
      "Producteur",
      "Code postal",
      "Coordonnée en X",
      "Coordonnée en Y",
      "geo_shape",
      "geo_point_2d"
    ],
    "description": "Reorder columns"
  }
```

In other words, the operation parameters simply remember what is the final order of the remaining columns after reorder.
As explained in #4055, this operation does not generalize well to other datasets, when it is used in the "Apply" dialog of the Undo/Redo tab, because any columns that were not in the original dataset but are present in the new one will be deleted by this operation.

### Proposed solution

We should find other ways to specify this operation so that the actual intent of the user is captured better.
It is difficult to come up with a precise specification, but as a litmus test, I would expect the following criteria to be satisfied:
* If I use the dialog above to delete columns only, without reordering the remaining columns, I would expect that the generalization of this operation only deletes the said columns, and leaves any other column (including those not present in the original dataset) untouched. This could be achieved by calling the `column-removal` operation instead, as it supports removing multiple columns at once after #5563.
* If I use the dialog to reorder columns only, without deleting any column, then I expect that the generalization of this operation will not delete any column in any other context. Additional columns not present in the original dataset should be placed in a *sensible* location in the resulting table. It is not a problem if this exact order is complicated to define, as the exact position of columns should generally not have a significant influence on operation workflows except in certain cases (such as the use of transpose operations).

### Alternatives considered

One could decide to break down the operation into multiple steps, applying the operations that remove or move a single column multiple times, to reach the desired state. Working on such a decomposition is likely useful to understand this issue better, but I would rather prefer that this dialog does not generate lengthy lists of steps in the project history.

### Additional context

Follow-up to #4055 and #5563.

**Comment by thadguidry** [comment]
I tend to agree with the proposal.  So are you for/against splitting the dialog into two menu items?
I am against splitting the dialog as it flows well, to reorder a few columns and delete/don't care about these other columns.
Behind the scenes, the current operation itself could be split into multiple operations.
Agree on non-lengthy lists but they are lengthy often enough when dealing with wide datasets, so.

**Comment by wetneb** [comment]
Yes I would keep the dialog as is.

**Comment by tfmorris** [comment]
A direct manipulation paradigm doesn't scale very well to deal with invisible elements and the undo/redo capability was designed to work, to the extent that it does, on projects **of the same shape**. To turn it into a full generalized automation language would require considerably more robust semantics.

The other approach would be to introduce more limitations and restrict the operations to things like "move column to beginning," "Move column to end," "Move column right," etc, but, while more logical programmatically,  that would be a much less nice user experience.

I think the number 1 aim should be to keep users from shooting themselves in the foot by unwittingly corrupting their data.

**Comment by sunilnatraj** [comment]
@wetneb Column reorder/delete being a bulk update command, maybe we can generalise the request to include the actual operation against each column, something like below. 

[
  {
    "op": "core/column-bulk-edit-action",
    "columns": [
      {
        "columnName": "filename",
        "operation": {
          "command": "reorder",
          "params": {
            "position": 0
          }
        }
      },
      {
        "columnName": "filepath",
        "operation": {
          "command": "reorder",
          "params": {
            "position": 1
          }
        }
      },
      {
        "columnName": "wikitext",
        "operation": {
          "command": "delete"
        }
      }
    ],
    "description": "Column bulk edit"
  }
]
